build_scripts/create_sysroot.sh

```diff
@@ -0,0 +1,106 @@
+#!/bin/bash
+
+# This script customizes the plutosdr-fw sysroot image to our needs.
+# NOTE: you only need this script if you want to rebuild the sysroot
+#       folder, i.e. after a new release of the plutosdr-fw!
+#       The script is automatically invoked by the other build scripts
+
+# Required plutosdr-fw version
+PLUTOSDR_FW_TAG="v0.32"
+
+
+SCRIPT_DIR=`pwd`/`dirname "$0"`
+BUILD_DIR=$SCRIPT_DIR/build
+PLUTOSDR_FW_DIR=$BUILD_DIR/plutosdr-fw
+SYSROOT_DIR=$BUILD_DIR/pluto-custom.sysroot
+SRC_DIR=$SCRIPT_DIR/..
+
+mkdir -p $BUILD_DIR
+cd $BUILD_DIR || exit 1
+mkdir -p .tmp_create_sysroot
+cd .tmp_create_sysroot || exit 1
+
+# Read current build status
+BUILD_STATUS=$BUILD_DIR/build_status
+touch $BUILD_STATUS
+source $BUILD_STATUS || exit 1
+
+if [[ $STATUS_DOWNLOAD_SYSROOT != "y" ]]
+then
+  echo "Could not locate existing sysroot dir. Downloading sources"
+  wget https://github.com/analogdevicesinc/plutosdr-fw/releases/download/$PLUTOSDR_FW_TAG/sysroot-$PLUTOSDR_FW_TAG.tar.gz
+  tar -xzf sysroot-$PLUTOSDR_FW_TAG.tar.gz || exit 1
+  mv staging pluto-custom.sysroot
+  mv pluto-custom.sysroot $BUILD_DIR/
+  echo "STATUS_DOWNLOAD_SYSROOT=y" >> $BUILD_STATUS
+else
+  echo "Found existing sysroot folder. Skip download."
+fi
+
+echo "Bulding libraries"
+
+if [[ $STATUS_SYSROOT_GOT_FFTW3F != "y" ]]
+then
+  echo "Downloading fftw3"
+  wget http://www.fftw.org/fftw-3.3.8.tar.gz
+  tar -xzf fftw-3.3.8.tar.gz
+
+  echo "Installing fftw3"
+  cd  fftw-3.3.8 || exit 1
+  ./configure arm --enable-float --enable-shared --host=arm-linux-gnueabihf --prefix="$SYSROOT_DIR/usr/" CC="arm-linux-gnueabihf-gcc --sysroot=$SYSROOT_DIR/"
+  make install || exit 1
```

Comment:
has to be done as sudo (as the other `make install` commands and ldconfig as well), so we should unlock sudo at the very beginning of the create_sysroot.sh (and the build_firmware.sh) script

Reply:
We are cross-compiling and `./configure --prefix` specifies the install directory to a path that does not require root privileges for write access. I do not think sudo is required (the script works on my machine).
It is not necessary to call `ldconfig` for the same reason, it only updates libs for the build machine, not the machine we are building for.

There is however a problem with _libfec_, since the `make install` target unnecessarily contains an ldconfig call, it requires sudo. For this case, I would suggest to run make without install (no sudo required) and copy the files manually to our sysroot directory.

Reply:
Thanks for the explanation!